### PR TITLE
Adding 2021-07-01 version in recoveryservices, recoveryservicessiterecovery folder to make it S360 compliant

### PR DESCRIPTION
**Pull request by deymadhumanti**
…veryservicessiterecovery RP same as recoveryservicesbackup

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Adding 2021-07-01 API version in recoveryservices, recoveryservicessiterecovery folder since this version is added recently in recoveryservicesbackup folder for S360 compliance.

No new changes are added here. I have just copied the existing json and example files in the new version folder (2021-07-01) to make it S360 compliant.

### CODE REVIEW

**Comment by pilor** [comment]
Signing off from API review side since there are no changes from previous version

**Comment by lirenhe** [comment]
@deymadhumanti, could you help to check the avocado errors? It seems there are a lot of unreferenced files.

**Comment by lirenhe** [comment]
There are also a couple of duplicate schema which would impact the sdk generation.

File: specification/recoveryservices/resource-manager/readme.md [hunk 1]
```diff
@@ -41,7 +41,17 @@ model-validator: true
...
+
+These settings apply only when `--tag=package-2021-06` is specified on the command line.
 
+```yaml $(tag) == 'package-2021-06'
+input-file:
+  - Microsoft.RecoveryServices/stable/2021-07-01/registeredidentities.json
+  - Microsoft.RecoveryServices/stable/2021-07-01/replicationusages.json
+  - Microsoft.RecoveryServices/stable/2021-07-01/vaults.json
+  - Microsoft.RecoveryServices/stable/2021-07-01/vaultusages.json
+```
```
Comment (qiaozha) [inline specification/recoveryservices/resource-manager/readme.md:54]: The default tag is package-2021-07 but here we add a tag package-2021-06

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/CheckNameAvailability_NotAvailable.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "parameters": {
+    "subscriptionId": "77777777-b0c6-47a2-b37c-d8e65a629c18",
+    "resourceGroupName": "resGroupBar",
+    "api-version": "2021-06-01",
```
Comment (lirenhe) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/CheckNameAvailability_NotAvailable.json:5]: do you need to update the api-version here?

Comment (lirenhe) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/CheckNameAvailability_NotAvailable.json:5]: It seems this issue happens in all your sample files.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/DeleteRegisteredIdentities.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "parameters": {
+    "subscriptionId": "77777777-d41f-4550-9f70-7708a3a2283b",
+    "resourceGroupName": "BCDRIbzRG",
+    "vaultName": "BCDRIbzVault",
+    "identityName": "dpmcontainer01",
+    "api-version": "2021-06-01"
```
Comment (nilgupta6) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/DeleteRegisteredIdentities.json:7]: Please change API version here

Comment (nilgupta6) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/DeleteRegisteredIdentities.json:7]: please change everywhere in example files

**Review by nilgupta6 — CHANGES_REQUESTED** [review]
@deymadhumanti : Can you please change API version in all example files and service.json to "2021-07-01" right now it is "2021-06-01" in all files. Please take care of this.

File: specification/recoveryservices/resource-manager/readme.md (hunk 1 above)
Comment (nilgupta6) [reply specification/recoveryservices/resource-manager/readme.md:54]: @deymadhumanti : please take care of this before merge

Comment (deymadhumanti) [reply specification/recoveryservices/resource-manager/readme.md:54]: Updated the version

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/CheckNameAvailability_NotAvailable.json (hunk 2 above)
Comment (deymadhumanti) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/CheckNameAvailability_NotAvailable.json]: updated the version in all swagger and example jsons

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/DeleteRegisteredIdentities.json (hunk 3 above)
Comment (deymadhumanti) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2021-07-01/examples/DeleteRegisteredIdentities.json]: Done

**Review by lirenhe — APPROVED** [review]

**Review by nilgupta6 — APPROVED** [review]
Approved

**Pull request merged**